site/content/concepts/device-notifications.md

```diff
@@ -0,0 +1,116 @@
++++
+title = "Device notifications (from 0.6 on)"
+weight = 415
++++
+
+*Business applications* can be informed by Hono about specific states of a device. This concept - refered to as
+ **Device notifications** - is described in detail here.
+ 
+ 
+
+{{% warning %}}
+This concept is currently part of active development and discussion.
+The content here summarizes the current discussion and is being prototyped.
+Please join our mailing list or consult the IRC channel in case you have any questions or want to participate.
+{{% /warning %}}
+
+<!--more-->
+
+For implementing specific behaviour *Business applications* need to know if
+
+- a device is currently ready to receive a command
+- a device is currently connected to a protocol adapter
+ 
+The first can serve e.g. to implement functionality like sending a command upstream to a device.
+
+In the following it is described how a *Business application* can be notified that a device is ready to receive a command. 
+This is referred to as `device command readiness`.
+
+## Device command readiness notification
+
+To notify an application about the command readiness of a device specific properties are set in a downstream message.
+
+This can be done
+
+- implicitly: when sending a downstream message for any reason (e.g. telemetry or event message)
+- explicitly: by sending a specific empty event (e.g. if a device does not have any other downstream message to send)
+
+
+The following AMQP 1.0 properties of the message are relevant for the notification:
+
+- the AMQP 1.0 property `creation-time` (which is automatically set by protocol adapters)
+- an additional application-property `ttd` (containing the amount of seconds the device stays connected) in the AMQP 1.0 message
+ 
+Please refer to the [Telemetry API]({{< relref "api/Telemetry-API.md" >}}) and the [Event API]({{< relref "api/Event-API.md" >}}) for further details.
+
+### Time based validation of notification 
+
+An application receiving a downstream message containing the notification relevant properties can verify at any point in
+time if a device is ready to receive a command by:
+
+- adding the values of the `creation-time` and the `ttd` properties to build a *valid until timestamp*
+- comparing the current time with the *valid until timestamp*
+
+If the current time is *before* the *valid until timestamp*, the notification shall be regarded as being currently valid.
```

Comment:
I do not think that we are talking about the *validity* of a notification. We are talking about the expiration of the ttd value.